exercise/api/views.py

```diff
@@ -87,7 +84,7 @@ class ExerciseViewSet(mixins.RetrieveModelMixin,
         url_path='grader',
         url_name='grader',
         methods=['get', 'post'],
-        permission_classes = GRADER_PERMISSION,
+        get_permissions = lambda: [JWTSubmissionCreatePermission()],
```

Comment:
The permissions system is stupid because of the inheritance: if you want to access an exercise , you also need to access a course instance. The grader only has access to that exercise and not the instance, so it doesn't work.